helix-core/src/test/java/org/apache/helix/integration/task/TestForceDeleteWorkflow.java

```diff
@@ -0,0 +1,433 @@
+import org.apache.helix.task.TaskCallbackContext;
+import org.apache.helix.task.TaskDriver;
+import org.apache.helix.task.TaskFactory;
+import org.apache.helix.task.TaskState;
+import org.apache.helix.task.TaskStateModelFactory;
+import org.apache.helix.task.Workflow;
+import org.apache.helix.task.WorkflowConfig;
+import org.apache.helix.task.WorkflowContext;
+import org.apache.helix.tools.ClusterSetup;
+import org.apache.helix.tools.ClusterVerifiers.BestPossibleExternalViewVerifier;
+import org.apache.helix.tools.ClusterVerifiers.ZkHelixClusterVerifier;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/*
+This test checks the functionality of the ForceDelete in various conditions.
+ */
+public class TestForceDeleteWorkflow extends ZkTestBase {
+  protected int _numNodes = 5;
+  protected int _startPort = 12918;
+  protected int _numPartitions = 20;
+  protected int _numReplicas = 3;
+  protected int _numDbs = 1;
+
+  protected ClusterControllerManager _controller;
+  protected HelixManager _manager;
+  protected TaskDriver _driver;
+
+  protected List<String> _testDbs = new ArrayList<>();
+
+  protected final String MASTER_SLAVE_STATE_MODEL = "MasterSlave";
+  protected final String CLUSTER_NAME = CLUSTER_PREFIX + "_" + getShortClassName();
+  protected MockParticipantManager[] _participants;
+
+  protected ZkHelixClusterVerifier _clusterVerifier;
+  private HelixAdmin _admin;
+
+  @BeforeClass
+  public void beforeClass() throws Exception {
+    super.beforeClass();
+    _participants = new MockParticipantManager[_numNodes];
+    _gSetupTool.addCluster(CLUSTER_NAME, true);
+
+    setupParticipants(_gSetupTool);
+    setupDBs();
+
+    for (int i = 0; i < _numNodes; i++) {
+      startParticipant(ZK_ADDR, i);
+    }
+
+    _manager = HelixManagerFactory.getZKHelixManager(CLUSTER_NAME, "Admin",
+        InstanceType.ADMINISTRATOR, ZK_ADDR);
+    _manager.connect();
+    _driver = new TaskDriver(_manager);
+
+    _clusterVerifier =
+        new BestPossibleExternalViewVerifier.Builder(CLUSTER_NAME).setZkClient(_gZkClient).build();
+
+    // start controller
+    String controllerName = CONTROLLER_PREFIX + "_0";
+    _controller = new ClusterControllerManager(ZK_ADDR, CLUSTER_NAME, controllerName);
+    _controller.syncStart();
+
+    Assert.assertTrue(_clusterVerifier.verifyByPolling());
+
+    _admin = _gSetupTool.getClusterManagementTool();
+  }
+
+  @AfterClass
+  public void afterClass() throws Exception {
+    if (_controller != null && _controller.isConnected()) {
+      _controller.syncStop();
+    }
+    if (_manager != null && _manager.isConnected()) {
+      _manager.disconnect();
+    }
+    for (int i = 0; i < _numNodes; i++) {
+      stopParticipant(i);
+    }
+    deleteCluster(CLUSTER_NAME);
+  }
+
+  protected void setupDBs() {
+    setupDBs(_gSetupTool);
+  }
+
+  protected void setupDBs(ClusterSetup clusterSetup) {
```

Comment:
Question: this seems like duplicate code. Don't we already have this somewhere in the codebase? Why are we duplicating?